src/buf/bounded.rs

```diff
@@ -0,0 +1,156 @@
+use super::{IoBuf, IoBufMut, Slice};
+
+use std::ops;
+
+/// A possibly bounded view into an owned [`IoBuf`] buffer.
+///
+/// Because buffers are passed by ownership to the runtime, Rust's slice API
+/// (`&buf[..]`) cannot be used. Instead, `tokio-uring` provides an owned slice
+/// API: [`.slice()`]. The method takes ownership of the buffer and returns a
+/// [`Slice`] value that tracks the requested range.
+///
+/// This trait provides a generic way to use buffers and `Slice` views
+/// into such buffers with `io-uring` operations.
+///
+/// [`.slice()`]: BoundedBuf::slice
+pub trait BoundedBuf: Sized + Unpin + 'static {
```

Comment:
Yes, surprisingly the `Self: Sized` bound is not needed here. I thought it must be somewhere for the `slice` method definition (like it was in `IoBuf` prior to this change, though that one had a default body), and also for `BoundedBuf::from_buf_bounds`, but the compiler does not require it. I think this is [an explanation](https://stackoverflow.com/a/54468427/2217151).

Since the `BoundedBuf` trait implementations are sealed in this crate, I think it's OK to remove the `Sized` supertrait. Regardless, it cannot be implemented on unsized types, e.g. changing the blanket impl's signature to `impl<T: ?Sized + IoBuf> BoundedBuf for T` will fail.
But I don't think it impedes any known usage either, since `IoBuf` is only implemented on sized types (including `&'static [u8]`).

Reply:
Thanks for the link. Very informative.